lightning-how-the-pieces-fit-together/README.md

```diff
@@ -14,9 +14,43 @@
 
 ## Discussion Questions
 
-1. How is the payment hash sent to the payer?
-1. Explain how the timeout mechanism works for in-flight HTLCs?
 1. How would an offline vending machine work?
-1. What are routing hints and why are they important for private channels? Do they leak privacy?
+
+### A walk through the layers of Lightning
+
 1. How does the LN-Penalty mechanism currently leak out of the update layer?
-1. If LN were to upgrade the update layer, to eltoo for example, would the network need to end-to-end upgrade?
+
+### The Update Layer
+
+1. How would switching to PTLCs affect the update layer?
+1. If LN were to upgrade the update layer, to Eltoo for example, would the network need to end-to-end upgrade?
+1. Would there ever be a reason to prefer LN-penalty when creating a channel between two Eltoo-capable nodes?
+1. Why are the second stage HTLC transactions not part of the commitment transaction?
+
+### The Transfer Layer
+
+1. Explain how the timeout mechanism works for in-flight HTLCs?
+1. How is the payment hash sent to the payer?
+1. Can both HTLCs and PTLCs coexist in a channel?
+1. What's are the required parts of invoices?
+
+### The Multihop Layer
+
+1. How can nodes that are not always online prevent loss of funds?
+1. What are the risks of forwarding below-dust payments?
+1. What is the relationship of the revocation keys for multiple subsequent commitment transactions? What would change if revocation keys were independent?
+1. How can you send a payment to a node without getting an invoice? What are the tradeoffs of this payment method?
+1. Why is a grace period added to the HTLC timeout?
+1. What are routing hints and why are they important for private channels? Do they leak privacy?
+
+### The other layers (base and transport)
```

Comment:
Probably remove if you don't have any questions to include

Reply:
I'll comment out the unused sections.